lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1551,7 +1551,7 @@ protected static function compute_preset_classes( $settings, $selector, $origins
 	 * @param string $selector Original selector.
 	 * @return string Scoped selector.
 	 */
-	protected static function scope_selector( $scope, $selector ) {
+	public static function scope_selector( $scope, $selector ) {
```

Comment:
I just noticed that we're making `scope_selector` a public API.

@oandregal, we had a similar discussion at https://github.com/WordPress/gutenberg/pull/46571#discussion_r1088049807. Do you think this would be a valuable method for third parties, worth maintaining the BC?

Reply:
The class is still private, so this shouldn't be considered a public API. We have other methods that are also public in this class, though unnecessarily too many, in my view. In https://github.com/WordPress/gutenberg/issues/45171 I list some that we need a public API for.

My concern in https://github.com/WordPress/gutenberg/pull/46571#discussion_r1088049807 was that we were _promoting_ or _expecting_ third-parties to use it. If we expect anything to be used by 3rd parties, we need to create a proper public API outside this private class.

In this particular case, I don't know that we expect 3rd parties to use it. If that's the case, I'd support extracting it to its own method as a utility used by block supports and this class. If we don't expect anyone else to use it, I feel less pressure to extract it (though it'd still be a good thing).

Reply:
It does help. Thank you, @oandregal!

Reply:
Exactly the class is internal. This function is not helpful as hopefully, third-parties never deal with low-level selectors, but if it turns out to be useful, we can extract it in a function. This PR is just fixing a regression that caused a crash when the functionality was originally merged. The function was already public and inside an internal class.